File: apps/api/app/models/voice.py
from pydantic import BaseModel, ConfigDict, Field


class VoiceSegment(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    frame_id: str = Field(alias="frameId")
    text: str
    mode: str
    role: str
    speaker: str | None = None
    voice_preset: str | None = Field(default=None, alias="voicePreset")
    audio_file: str | None = Field(default=None, alias="audioFile")
    transcript: str | None = None
    duration_ms: int | None = Field(default=None, alias="durationMs")

File: packages/schema/src/voice.ts
import { z } from "zod";

export const voiceSchema = z.object({
  id: z.string().min(1),
  frameId: z.string().min(1),
  text: z.string(),
  mode: z.enum(["tts", "record", "skip"]),
  role: z.enum(["narrator", "character"]),
  speaker: z.string().min(1).nullish(),
  voicePreset: z.string().min(1).nullish(),
  audioFile: z.string().min(1).nullish(),
  transcript: z.string().min(1).nullish(),
  durationMs: z.number().int().positive().nullish(),
});

export type VoiceSegment = z.infer<typeof voiceSchema>;
